sympy/printing/repr.py

```diff
@@ -144,6 +144,14 @@ def _print_EmptySequence(self, expr):
     def _print_list(self, expr):
         return "[%s]" % self.reprify(expr, ", ")
 
+    def _print_dict(self, expr):
+        sep = ", "
+        dict_kvs = ["%s: %s" % (self.doprint(key), self.doprint(value)) for key, value in expr.items()]
+        return "{%s}" % sep.join(dict_kvs)
+
+    def _print_set(self, expr):
+        return "{%s}" % self.reprify(expr, ", ")
```

Comment:
This is wrong, and prints empty sets as empty dictionaries
```suggestion
        if not expr:
            return "set()"
        else:
            return "{%s}" % self.reprify(expr, ", ")
```